core/lib/zksync_core/src/lib.rs

```diff
@@ -174,6 +175,20 @@ pub async fn genesis_init(
         },
     )
     .await?;
+
+    if wait_for_set_chain_id {
+        genesis::save_set_chain_id_tx(
+            eth_client_url,
+            contracts_config.diamond_proxy_addr,
+            contracts_config
+                .state_transition_proxy_addr
+                .expect("state_transition_proxy_addr is not set, but needed for genesis"),
```

Comment:
Nit: Since this function propagates errors, `.context(_)` would be more idiomatic here.